src/pull_module/gguf_downloader.cpp

```diff
@@ -66,19 +97,29 @@ Status GGUFDownloader::downloadModel() {
         SPDLOG_ERROR("Path {} escape with .. is forbidden.", this->downloadPath);
         return StatusCode::PATH_INVALID;
     }
-
-    if (std::filesystem::is_directory(this->downloadPath) && !this->hfSettings.overwriteModels) {
-        SPDLOG_DEBUG("Path already exists on local filesystem. Not downloading to path: {}", this->downloadPath);
-        return StatusCode::OK;
+    if (!this->hfSettings.ggufFilename.has_value() || this->hfSettings.ggufFilename->empty()) {
+        SPDLOG_ERROR("GGUF filename must be specified for GGUF download type, and shouldn't be empty.");
+        return StatusCode::INTERNAL_ERROR;
     }
-    std::filesystem::create_directories(this->downloadPath);
-    ovms::Status status;
+    auto status = checkIfOverwriteAndRemove(this->hfSettings, this->downloadPath);
     if (!status.ok()) {
         return status;
     }
-    status = checkIfOverwriteAndRemove(this->hfSettings, this->downloadPath);
-    if (!status.ok()) {
-        return status;
+    if (!std::filesystem::is_directory(this->downloadPath)) {
+        if (!std::filesystem::create_directories(this->downloadPath)) {
+            SPDLOG_ERROR("Failed to create model directory: {}", this->downloadPath);
+            return StatusCode::PATH_INVALID;
+        }
+    }
+    if (!this->hfSettings.overwriteModels) {
+        auto statusOrBool = checkIfAlreadyExists(this->hfSettings, this->downloadPath);
+        if (std::holds_alternative<Status>(statusOrBool)) {
+            return std::get<Status>(statusOrBool);
+        }
+        if (std::get<bool>(statusOrBool)) {
+            SPDLOG_DEBUG("Model files already exist and overwrite is disabled, skipping download for model: {}", this->hfSettings.sourceModel);
```

Comment:
Log files not directory: this->hfSettings.sourceModel

Reply:
Visible here:
https://github.com/openvinotoolkit/model_server/pull/3626/files/e96891f2454b80e276fe8238e2a1fb4cfcd0c3c0#diff-647cdfdfc0834d6361870da07fa4a96e18686472e3a638c0ae7de8920d7ca7a8R74 on DEBUG level